layers/core_checks/cc_device_memory.cpp

```diff
@@ -824,6 +833,44 @@ bool CoreChecks::ValidateMemoryTypes(const vvl::DeviceMemory &mem_info, const ui
     return skip;
 }
 
+bool CoreChecks::ValidateTileMemorySizeInfo(const VkTileMemorySizeInfoQCOM pCreateInfo, const Location &loc) const {
+    bool skip = false;
+
+    uint32_t memoryHeapCount = phys_dev_mem_props.memoryHeapCount;
+    uint64_t largestHeapSize = 0;
+    for (uint32_t i = 0; i < memoryHeapCount; i++) {
```

Comment:
```suggestion
    for (uint32_t i = 0; i < phys_dev_mem_props.memoryHeapCount; i++) {
```

or 

`const uint32_t memory_heap_count = phys_dev_mem_props.memoryHeapCount;`